pkg/controller/sriovnetwork/sriovnetwork_controller.go

```diff
@@ -142,32 +144,47 @@ func (r *ReconcileSriovNetwork) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, err
 	}
 
-	// Check if this NetworkAttachmentDefinition already exists
-	found := &netattdefv1.NetworkAttachmentDefinition{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: netAttDef.Name, Namespace: netAttDef.Namespace}, found)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			reqLogger.Info("Creating a new NetworkAttachmentDefinition", "Namespace", netAttDef.Namespace, "Name", netAttDef.Name)
-			err = r.client.Create(context.TODO(), netAttDef)
+	if namespace, ok = lastNetworkNamespace[instance.GetUID()]; ok {
+		if namespace == netAttDef.GetNamespace() {
+			// Check if this NetworkAttachmentDefinition already exists
+			found := &netattdefv1.NetworkAttachmentDefinition{}
+			err = r.client.Get(context.TODO(), types.NamespacedName{Name: netAttDef.Name, Namespace: namespace}, found)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+			reqLogger.Info("NetworkAttachmentDefinition already exists, updating")
+			netAttDef.SetResourceVersion(found.GetResourceVersion())
+			err = r.client.Update(context.TODO(), netAttDef)
+			if err != nil {
+				reqLogger.Error(err, "Couldn't update NetworkAttachmentDefinition", "Namespace", netAttDef.Namespace, "Name", netAttDef.Name)
+				return reconcile.Result{}, err
+			}
+			// NetworkAttachmentDefinition updated successfully - don't requeue
+			return reconcile.Result{}, nil
+		} else {
+			err = r.client.Delete(context.TODO(), &netattdefv1.NetworkAttachmentDefinition{
+			    ObjectMeta: metav1.ObjectMeta{
+					Name: netAttDef.GetName(), 
```

Comment:
nit: there is extra space at the end of this line (`Name: netAttDef.GetName(),`).

Reply:
fixed